### ISSUE TITLE
Better release management: Provide release notes and publish github releases

### ISSUE THREAD

**Issue by StarpTech** [traefik/traefik-helm-chart#265]
Hi,
this is the official traefik helm chart. I'd expect that the same releasing strategy is used as in traefik. Currently, it's very hard to track changes.

Additionally, I propose to not release from the master after every merge. There is a high chance to break systems like https://github.com/traefik/traefik-helm-chart/commit/98f4575d75b91c6e739ebc0e75c720c998f7dd38#r41556509

**Comment by ldez** [comment]
Related to #175, #250

**Comment by ldez** [comment]
Fixed by ba98e364cf82d3c0d179b321aa668d53dc10e030

We now create a tag for each version.

**Comment by StarpTech** [comment]
@ldez not sure how it will fix the issue. This will only create a tag on untagged helm versions. The whole story is https://helm.sh/docs/community/release_checklist/

**Comment by ldez** [comment]
It's the release process of Helm itself, it's not related to Helm Chart.

**Comment by StarpTech** [comment]
My mistake. I mean we should produce proper release notes.

**Comment by ldez** [comment]
The lifecycle of a Helm chart is not the same as the product handle by the Helm chart.
The changes in the Helm chart are globally not related to the product changes.
Every change required a change of the Helm Chart version, it's the best practice.

So it's not possible to correlate Traefik version and the Helm chart version.

**Comment by ldez** [comment]
The changelog can be obtained by using the compare feature of Github.

ex: https://github.com/containous/traefik-helm-chart/compare/v8.10.0...v9.1.0

**Comment by StarpTech** [comment]
If we can make a release "new version" we can also create a changelog about the changes.

> ex: v8.10.0...v9.1.0

Yes, this one way but https://github.com/traefik/traefik/releases is much more readable and expressive. This is totally up to you. If you don't want to do this it's fine.

**Comment by ldez** [comment]
As each commit is a version the changelog is the same as git history.
Create a Github Release to add a one-line changelog seems overkill, so we will not automate this part.

But now, with the new tag system, we will be able, when it's relevant, to create manually a GitHub Release.
For example, to explain the breaking changes.

**Comment by StarpTech** [comment]
I wasn't aware of that. This feels odd. Producing a tag for every commit is overkill. Why did you do this?
Releases aka "tags" aka "Github Releases" should be well prepared and most of the things can be automated.

**Comment by StarpTech** [comment]
Let me rephrase. Currently, we will create a tag for every new helm version, right? That means we can also automate the Github Release with a changelog.

**Comment by ldez** [comment]
There is a huge difference between a simple tag and GitHub Release with changelog.
We will not automate this part.

**Comment by StarpTech** [comment]
That's right but the issue was exactly about that.
